proposals/dictionary-expressions.md

```diff
@@ -55,178 +55,86 @@ Choices here would have implications regarding potential syntactic ambiguities,
+2. Introduction of the *dictionary type*. *Dictionary types* are types that are similar to the existing *collection types*, with the additional requirements that they have an *element type* of some `KeyValuePair<TKey, TValue>` *and* have an indexer `TValue this[TKey] { ... }`. The former requirement ensures that `List<T>` is not considered a dictionary type, as its element type is `T` not `KeyValuePair<,>`. The latter requirement ensures that `List<KeyValuePair<int, string>>` is also not considered a dictionary type, with its `int`-to-`KeyValuePair<int, string>` indexer (instead of an `int`-to-`string` indexer). `Dictionary<TKey, TValue>` passes both requirements.
 
-### Answered question 2
+    As such, if the target type for the collection expression *is* a *dictionary* type, then all `KeyValuePair<,>` produced by `expression_element` or `spread_element` elements will be changed to use the indexer to assign into the resultant dictionary. Any `key_value_pair_element` will use that indexer directly as well.  For example:
 
-Can you spread a *non dictionary type* when producing a dictionary type'd value.  For example:
+    ```c#
+    Dictionary<string, int> nameToAge = ["mads": 21, existingDict.MaxPair(), .. otherDict];
 
-```c#
-Dictionary<string, int> nameToAge = ["mads": 21, .. existingListOfKVPS];
-``` 
-
-**Resolution:** *Spread elements* of key-value pair collections will be supported in dictionary expressions. [LDM-2024-03-11](https://github.com/dotnet/csharplang/blob/main/meetings/2024/LDM-2024-03-11.md#conclusions)
-
-### Answered question 3
+    // would be rewritten similar to:
 
-How far do we want to take this KeyValuePair representation of things? Do we allow *key value pair elements* when producing normal collections? For example, should the following be allowed:
+    Dictionary<string, int> __result = new();
+    __result["mads"] = 21;
 
-```c#
-List<KeyValuePair<string, int>> = ["mads": 21];
-```
+    // Note: the below casts must be legal for the dictionary
+    // expression to be legal
+    var __t1 = existingDict.MaxPair();
+    __result[(string)__t1.Key] = (int)__t1.Value;
 
-**Resolution:** *Key value pair elements* will be supported in collection expressions for collection types that have a key-value pair element type. [LDM-2024-03-11](https://github.com/dotnet/csharplang/blob/main/meetings/2024/LDM-2024-03-11.md#conclusions)
+    foreach (var __t2 in otherDict)
+        __result[(string)__t2.Key] = (int)__t2.Value;
+    ```
 
-### Answered question 4
+3. Alignment of the rules for assigning to *dictionary types* with the rules for assigning to *collection types*, just requiring aspects such as *element* and *iteration types* to be some `KeyValuePair<,>`.  *However*, with the rules extended such that that `KeyValuePair<,>` type itself is relatively transparent, and instead the rule is updated to work on the underlying `TKey` and `TValue` types.
```

Comment:
added this new intuition.  Specifically that while `KVP<>` is the element type, we want to operate more on the level of Keys/Values.   I give an example below this of why that is desirable.